src/routes/bounce_rules_activity.jsx

```diff
@@ -0,0 +1,228 @@
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
+    return rules.filter(
+      rule =>
+        rule.bounce_action.toLowerCase().includes(searchToken) ||
+        rule.description.toLowerCase().includes(searchToken)
+    );
+  }
+
+  paginate(activityLog) {
+    const { currentActivityPageIndex, rulesToShow } = this.state;
+    const ruleStartIndex = (currentActivityPageIndex - 1) * rulesToShow;
+    const ruleEndIndex =
+      (currentActivityPageIndex - 1 * currentActivityPageIndex + rulesToShow) *
+      currentActivityPageIndex;
+    return activityLog.slice(ruleStartIndex, ruleEndIndex);
+  }
+
+  updatePageIndex(e) {
+    const newIndex = parseInt(e.currentTarget.getAttribute("value"), 10);
+    this.setState(prevState => {
+      const isPageIndexUpdated =
+        prevState.currentActivityPageIndex !== newIndex;
+      return {
+        currentActivityPageIndex: isPageIndexUpdated
+          ? newIndex
+          : prevState.currentActivityPageIndex,
+      };
+    });
+  }
+
+  handlePrevClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex - 1,
+    }));
+  }
+
+  handleNextClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex + 1,
+    }));
+  }
+
+  isDuplicate(searchCategory, searchToken) {
+    let found = false;
+    const { filterOptions } = this.state;
+    filterOptions.filter(filter => {
+      if (
+        filter.searchCategory === searchCategory &&
+        filter.searchToken === searchToken
+      ) {
+        found = true;
+      }
+      return false;
+    });
+    return found;
+  }
+
+  addFilter() {
+    const { searchCategory, searchToken } = this.state;
+    if (!searchCategory || !searchToken) {
+      this.setState({
+        invalidFilter: true,
+      });
+      return;
+    }
+    if (!this.isDuplicate(searchCategory, searchToken)) {
+      this.setState(prevState => ({
+        invalidFilter: false,
+        filterOptions: [
+          ...prevState.filterOptions,
+          { searchCategory, searchToken },
+        ],
+        searchToken: "",
+      }));
+    } else {
+      this.setState({
+        invalidFilter: true,
+      });
+    }
+  }
+
+  removeFilter(filter) {
+    const { filterOptions } = this.state;
+    const newFilter = [...filterOptions];
+    const index = newFilter.indexOf(filter);
+    newFilter.splice(index, 1);
+    this.setState({
+      filterOptions: [...newFilter],
+    });
+  }
+
+  handleBounceTabClicked() {
+    this.setState({
+      isActivityLogTab: false,
+      isBounceRulesTab: true,
+    });
+  }
+
+  handleActivityTabClicked() {
+    this.setState({
+      isActivityLogTab: false,
```

Comment:
looks like `isActivityLogTab` should be set to `true` and `isBounceRulesTab` should be set to `false`